node/endpoint-handler.js

```diff
@@ -77,7 +78,7 @@ TChannelEndpointHandler.prototype.waitForArg1 = function waitForArg1(req, buildR
 TChannelEndpointHandler.prototype.handleArg1 = function handleArg1(req, buildResponse, arg1) {
     var self = this;
     var name = String(arg1);
-    var handler = self.endpoints[name];
+    var handler = coerceRequestHandler(self.endpoints[name], self);
```

Comment:
coerces on every request rather than once at registration time, please make it once